[WFLY-16126/WFLY-16250]: Problematic Language usage for messaging subsystem

ehsavoie opened the pull request:
2 commits for the 2 issues:
 - WFLY-16126: allowlist/blocklist
 - WFLY-16205: primary/secondary

Jira: 
* https://issues.redhat.com/browse/WFLY-16126
* https://issues.redhat.com/browse/WFLY-16205

ehsavoie committed:
[WFLY-16126]: [allowlist/blocklist] Problematic Language usage deprecation and replacement in messaging.

* Replacing problematic language parameters
* Addign transformers

Jira: https://issues.redhat.com/browse/WFLY-16126

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/ConnectionFactoryAttributes.java:
```diff
@@ -464,6 +489,8 @@ interface Common {
                 create(THREAD_POOL_MAX_SIZE, "threadPoolMaxSize", true),
                 create(GROUP_ID, "groupID", true),
+                create(DESERIALIZATION_ALLOWLIST, "deserializationWhiteList", true),
+                create(DESERIALIZATION_BLOCKLIST, "deserializationBlackList", true),
```
I believe the string constants here are incorrect.

ehsavoie replied:
No, that's the resource adapter property which hasn't changed since it is an Artemis class

bstansberry commented on messaging-activemq/subsystem/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties:
```diff
@@ -167,8 +167,10 @@ connection-factory.connection-ttl=The connection ttl.
 connection-factory.consumer-window-size=The consumer window size.
+connection-factory.deserialization-allow-list=A list of class names (separated by whitespaces) that are allowed to be used in serialization of Jakarta Messaging ObjectMessage.
 connection-factory.deserialization-black-list=A list of class names (separated by whitespaces) that are not allowed to be used in serialization of Jakarta Messaging ObjectMessage.
 connection-factory.deserialization-black-list.deprecated=Will be replaced by a differently-named attribute in a later release.
```
These .deprecated messages should be updated to reference the replacement attributes.

bstansberry commented on messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/jms/ConnectionFactoryAttributes.java:
```diff
@@ -214,6 +219,16 @@ interface Common {
                 .setAttributeMarshaller(AttributeMarshaller.STRING_LIST)
                 .setDeprecated(ModelVersion.create(13, 0, 0), false)
+                .setAlternatives("deserialization-allow-list")
+                .setRestartAllServices()
```
Did you try adding .setFlags(AttributeAccess.Flag.ALIAS)  to this and the other deprecated one?

That will suppress this from read-resource and read-attribute-group output unless the op includes the 'aliases=true' param.

ehsavoie replied:
I did it when registering the attribute in the ResourceDefinition: https://github.com/wildfly/wildfly/pull/15603/files#diff-a3324164caaccf163c3d99b6c312a49217e3d66db679305d55ab59ac8a4884c6R87

bstansberry commented on messaging-activemq/subsystem/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties:
```diff
@@ -779,8 +781,10 @@ pooled-connection-factory.credential-reference.store=The name of the credential
 pooled-connection-factory.credential-reference.clear-text=Secret specified using clear text (check credential store way of supplying credential/secrets to services)
+pooled-connection-factory.deserialization-allow-list=A list of class names (separated by whitespaces) that are allowed to be used in serialization of Jakarta Messaging ObjectMessage.
 pooled-connection-factory.deserialization-black-list=A list of class names (separated by whitespaces) that are not allowed to be used in serialization of Jakarta Messaging ObjectMessage.
 pooled-connection-factory.deserialization-black-list.deprecated=Will be replaced by a differently-named attribute in a later release.
```
@ehsavoie This pair of 'deprecated' need messages need updating as well.

ehsavoie committed:
[WFLY-16205]:  [primary/secondary] Problematic Language usage deprecation and replacement in messaging.

* Remaming the problematic resources in HA:
  - ReplicationMasterDefinition
  - ReplicationSlaveDefinition
  - SharedStoreMasterDefinition
  - SharedStoreSlaveDefinition

Jira: https://issues.redhat.com/browse/WFLY-16205

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented:
Thanks @ehsavoie
